corehq/apps/users/tests/test_device_rate_limiter.py

```diff
@@ -1,77 +1,100 @@
-from django.test import SimpleTestCase, override_settings
+from django.test import TestCase
 from freezegun import freeze_time
 
 from corehq.apps.cloudcare.const import DEVICE_ID as CLOUDCARE_DEVICE_ID
-from corehq.apps.users.device_rate_limiter import device_rate_limiter
+from corehq.apps.users.device_rate_limiter import DEVICE_LIMIT_PER_USER_KEY, device_rate_limiter
+from corehq.project_limits.models import SystemLimit
 from corehq.tests.pytest_plugins.reusedb import clear_redis
-from corehq.util.test_utils import flag_enabled
+from corehq.util.test_utils import flag_disabled, flag_enabled
 
 
 @freeze_time("2024-12-10 12:05:43")
-@override_settings(DEVICE_LIMIT_PER_USER=1)
-@override_settings(ENABLE_DEVICE_RATE_LIMITER=True)
-class TestDeviceRateLimiter(SimpleTestCase):
+class TestDeviceRateLimiter(TestCase):
 
     domain = 'device-rate-limit-test'
 
     def setUp(self):
+        SystemLimit.objects.create(key=DEVICE_LIMIT_PER_USER_KEY, limit=1)
         self.addCleanup(clear_redis)
 
+    @flag_enabled("DEVICE_RATE_LIMITER")
```

Comment:
Can this decorator not be applied to the class? If not, wonder how hard it would be to change it to work at the class level?

Reply:
it works at the class level, but then `@flag_disabled` didn't work for the one test where I wanted to use it since the patch had already been applied for the `@flag_enabled` decorator. I briefly looked to see if it would be an easy fix but came up empty handed and decided to just do this instead.

Reply:
:slightly_frowning_face:

Reply:
I added these decorators originally and I vaguely recall also running into that shortcoming and not seeing an easy fix.  Would love to see that addressed if anyone can figure it out.

Reply:
Added it as a hackathon idea for tomorrow 🤷🏻

Reply:
Shoutout @millerdev https://github.com/dimagi/commcare-hq/pull/35818

Reply:
2d5aea65bd88ca69793c69f27ce2f12afc3334e6